sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
The previous code was more obviously correct, while getting the bulk of the benefit of reducing allocations. I'm particularly worried about the interaction between reusing the buffer between requests and `readBuffer.getString`. 

What affect did this change have on the `Select1_PGWire` benchmark? Making a perf change without measurement is a good way to accidentally move backwards.